chore(upgrade.md): improve wording (#13046)
kumahq/kuma: UPGRADE.md

## Upgrade to `2.10.x`

### API Server behaviour changes

> Successful `DELETE` or `PUT` requests without any warnings will now get "content-type: application/json" in response header and empty json as response content.

Reviewer: ```suggestion
Successful `DELETE` or `PUT` requests without warnings will now include "content-type: application/json" in the response header and return an empty JSON object as the response body.
```

> Successful `DELETE` or `PUT` requests without warnings will now include "content-type: application/json" in the response header and return an empty JSON object as the response body.

Reviewer 2: ```suggestion
The response of successful `DELETE` or `PUT` requests without warnings will now include "content-type: application/json" in the header and return an empty JSON object as the body.
```

> * Characters start with an alphabetic character.
> * Characters end with an alphanumeric character.

### On Universal, a MeshService is no longer generated for any dataplane that has an invalid `kuma.io/service` tag

> When using `MeshService`, we were automatically generating `MeshService` resources in Universal mode. However, due to stricter resource validation, we have decided not to generate the `MeshService` object when a dataplane has a `kuma.io/service` that is not [RFC 1035](https://www.rfc-editor.org/rfc/rfc1035.html) compliant.

Reviewer: ```suggestion
When using `MeshService`, we were automatically generating `MeshService` resources in Universal mode. However, due to stricter resource validation, we have decided not to generate the `MeshService` resource when a dataplane has a `kuma.io/service` that is not [RFC 1035](https://www.rfc-editor.org/rfc/rfc1035.html) compliant.
```

> If you want `MeshService` objects to be autogenerated by the control plane on Universal, update the `kuma.io/service` tags to valid names. Otherwise, you must create them manually.

Reviewer: ```suggestion
If you want the control plane on Universal to autogenerate `MeshService` resources, update the `kuma.io/service` tags to valid names. Otherwise, you must create them manually.
```

#### Deprecation of `healthyPanicThreshold` for `MeshHealthCheck`

> The field `healthyPanicThreshold` is deprecated on `MeshHealthCheck` and will be removed in a future release. It is now added to `MeshCircuitBreaker` policy. 

Reviewer: ```suggestion
The `healthyPanicThreshold` field in the `MeshHealthCheck` policy is deprecated and will be removed in a future release. It has been moved to the `MeshCircuitBreaker` policy.
```
